### ISSUE TITLE
[proposal] French talk to promote Rust at Belgian University (ECAM)

### ISSUE THREAD

**Issue by azerupi** [rust-community/talks#17]
I am an Engineering student in a Belgian university called [ECAM](http://www.vinci.be/fr-be/ecam/Pages/Accueil.aspx) and I proposed to give a talk about Rust. My computer science teacher is very eager to hear about it :wink: 

**The talk will be in French, so are the slides and the outline**

There is not date set, yet. So no pressure, though I would like to give it before my exam period (may - june).

## Audience

The audience will be comprised of:

- Bachelor students familiar with mostly Python
- Master students familiar with Python and C and a good background in Electronics
- Teachers mostly with a background in Electronics and (old) programming languages :wink: 
- Maybe a couple of lost souls who stumbled upon the announcement?

There will be no C / C++ experts (or only few) but a good part of the audience should already have experience with it. Most of them should have a strong background in electronics.

## Goal

The goal of the presentation is to promote Rust. A lot of students and teachers have never heard of it and I hope I can spark their curiosity. I want to explain it's strengths and compare it to what they know (C / C++ and Python). 

Given that most of the audience will have a pretty strong electronics background it could be interesting to briefly talk about the status of Rust in embedded systems and the plans for the future. But I would need help for that, because I don't know enough about it and wouldn't want to propagate false information. (I only know about the possibility to compile to ARM and the [avr-LLVM](https://github.com/avr-llvm/llvm) project)

## Topics I want to cover

- Memory safety with code example in C++ and Rust to compare both
- Ownership & Borrowing (only briefly)
- Concurrency
- Cargo
- Rust on embedded devices?
- ...

This is not set in stone and is open to suggestions. I already have a draft for memory safety + ownership & borrowing.

## Presentation

The actual presentation can be found [here](https://github.com/azerupi/fr-rust-presentation) (still a WIP). It's based on [reveal.js](http://lab.hakim.se/reveal-js/#/). The README explains how you can run it locally but I haven't tested it on a fresh machine yet..

I have written down an "outline" (more like the actual text) that you can read [here](https://github.com/azerupi/fr-rust-presentation/blob/master/Presentation.md)

I also wrote a utility in Rust that accepts POST request containing code, compiles it and returns the result. For now it works with C++ and Rust. I use it to have code output directly integrated into my slides. I call it [codeplatter](https://github.com/azerupi/codeplatter). 

------

I am not sure what I should ask here, but any feedback is appreciated and feel free to propose new ideas to explore! :smile:

**Comment by erickt** [comment]
This sounds great! Unfortunately I've long forgotten how to read french, but perhaps @pnkfelix, who is one of the organizers of [Rust-Paris](http://www.meetup.com/Rust-Paris/), might be able to help, or know someone who would.

Would you be interested in submitting your presentation here so that other french speakers could use your slides in the future?

**Comment by erickt** [comment]
I've brought this up on the [#rust-fr irc.mozilla.org channel](https://client00.chat.mibbit.com/?server=irc.mozilla.org&channel=%23rust-fr).

**Comment by azerupi** [comment]
>Would you be interested in submitting your presentation here so that other french speakers could use your slides in the future?

Of course! For now they are accessible [in my repo](https://github.com/azerupi/fr-rust-presentation) but once they are finished I will gladly submit them here. Just tell me how / where you want me to put them :)

>Unfortunately I've long forgotten how to read french

I can also use help from non-french speaking people :wink: 

I want to briefly talk about the strengths of Rust for concurrency with a couple of to the point examples. I have watched [Steve's Presentation](http://oredev.org/2015/sessions/concurrency-in-rust) several times already, but I can't fit the examples he uses into my presentation because there would not be enough time... 

Do you think I can find one or two examples where Rust shines, that I can explain in 5 - 10 minutes? Or is that a little too optimistic?

**Comment by Manishearth** [comment]
could you publish the built presentation to gh-pages so that others can view it without building?

**Comment by tomaka** [comment]
*Coming from #rust-fr*

I quickly read the talk. Many parts of it look inspired from other english talks given, like the ones from steveklabnik. That's a good point since at least you know that you are using a well-proven approach.

Other than that, it looks a bit long to me. I'm not that experienced in talks but it looks maybe one hour long. I'm not sure if that's the intended length.

**Comment by azerupi** [comment]
>could you publish the built presentation to gh-pages so that others can view it without building?

I can, the examples are not gonna run though :)

**Comment by azerupi** [comment]
>Other than that, it looks a bit long to me. I'm not that experienced in talks but it looks maybe one hour long. I'm not sure if that's the intended length.

I have measured a dry run from the start until the explanation of the Rust error message and it was about 17 minutes long. I made sure not to talk to fast and take some short pauses where I would have them in a real scenario. Of course this assumes I am not interrupted etc. 

For an estimation of a more realistic run time I would add a 10 min margin, which would make it about 30 minutes.

I don't have strict time constraints but I think aiming for a 45-60 min presentation would be the sweet spot. So that leaves me 15-20 minutes of remaining time to cover the other things.
